Error Building with QT5

It looks like a recent change has broken the building against QT5
Now I'm getting the following errors:

> 60>------ Build started: Project: opencv_highgui, Configuration: Release x64 ------
> 60>Generating src/moc_window_QT.cpp
> 60>Generating qrc_window_QT.cpp
> 60>Building Custom Rule D:/opencv/opencv/modules/highgui/CMakeLists.txt
> 60>CMake does not need to re-run because D:/opencv/opencv/build/modules/highgui/CMakeFiles/generate.stamp is up-to-date.
> 60>opencv_highgui_pch.cpp
> 60>window.cpp
> 60>roiSelector.cpp
> 60>window_QT.cpp
> 60>moc_window_QT.cpp
> 60>d:\opencv\opencv\modules\highgui\src\window.cpp(472): error C2664: 'CvFont cvFontQt(const char *,int,CvScalar,int,int,int)': cannot convert argument 3 from 'cv::Scalar' to 'CvScalar'
> 60>d:\opencv\opencv\modules\highgui\src\window.cpp(472): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
> 60>d:\opencv\opencv\modules\highgui\src\window.cpp(479): error C2440: 'initializing': cannot convert from 'const cv::Mat' to 'CvMat'
> 60>d:\opencv\opencv\modules\highgui\src\window.cpp(479): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
> 60>d:\opencv\opencv\modules\highgui\src\window.cpp(480): error C2664: 'void cvAddText(const CvArr *,const char *,CvPoint,CvFont *)': cannot convert argument 3 from 'cv::Point' to 'CvPoint'
> 60>d:\opencv\opencv\modules\highgui\src\window.cpp(480): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
> 60>d:\opencv\opencv\modules\highgui\src\window.cpp(486): error C2664: 'CvFont cvFontQt(const char *,int,CvScalar,int,int,int)': cannot convert argument 3 from 'cv::Scalar' to 'CvScalar'
> 60>d:\opencv\opencv\modules\highgui\src\window.cpp(486): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
> 60>d:\opencv\opencv\modules\highgui\src\window.cpp(487): error C2440: 'initializing': cannot convert from 'const cv::Mat' to 'CvMat'
> 60>d:\opencv\opencv\modules\highgui\src\window.cpp(487): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
> 60>d:\opencv\opencv\modules\highgui\src\window.cpp(488): error C2664: 'void cvAddText(const CvArr *,const char *,CvPoint,CvFont *)': cannot convert argument 3 from 'cv::Point' to 'CvPoint'
> 60>d:\opencv\opencv\modules\highgui\src\window.cpp(488): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
> 60>qrc_window_QT.cpp
> 60>d:\opencv\opencv\modules\highgui\src\window_qt.cpp(106): warning C4838: conversion from 'int' to 'float' requires a narrowing conversion
> 60>Done building project "opencv_highgui.vcxproj" -- FAILED.

I'm building on Windows 10 with Visual Studio 2017.

--- a/modules/highgui/src/window.cpp
+++ b/modules/highgui/src/window.cpp
@@ -469,23 +469,23 @@ CV_IMPL void cvUpdateWindow(const char*)
 
 cv::QtFont cv::fontQt(const String& nameFont, int pointSize, Scalar color, int weight, int style, int spacing)
 {
-    CvFont f = cvFontQt(nameFont.c_str(), pointSize,color,weight, style, spacing);
+    CvFont f = cvFontQt(nameFont.c_str(), pointSize, cvScalar(color), weight, style, spacing);
     void* pf = &f; // to suppress strict-aliasing
     return *(cv::QtFont*)pf;
 }
 
 void cv::addText( const Mat& img, const String& text, Point org, const QtFont& font)
 {
-    CvMat _img = img;
-    cvAddText( &_img, text.c_str(), org, (CvFont*)&font);
+    CvMat _img = cvMat(img);
+    cvAddText( &_img, text.c_str(), cvPoint(org), (CvFont*)&font);
 }
 
 void cv::addText( const Mat& img, const String& text, Point org, const String& nameFont,
         int pointSize, Scalar color, int weight, int style, int spacing)
 {
-    CvFont f = cvFontQt(nameFont.c_str(), pointSize,color,weight, style, spacing);
-    CvMat _img = img;
-    cvAddText( &_img, text.c_str(), org, &f);
+    CvFont f = cvFontQt(nameFont.c_str(), pointSize, cvScalar(color), weight, style, spacing);
+    CvMat _img = cvMat(img);
+    cvAddText( &_img, text.c_str(), cvPoint(org), &f);
 }
 
 void cv::displayStatusBar(const String& name,  const String& text, int delayms)

